stdlib/format.mli

```diff
@@ -107,6 +103,446 @@
 
 *)
 
+(** {1 Tutorial}
+
+{2 Principles}
+
+Line breaking is based on three concepts:
+
+{ul {- {b boxes} : a box is a logical pretty-printing unit, which
+    defines a behaviour of the pretty-printing engine to display the
+    material inside the box.}
+    {- {b break hints}: a break hint is a directive to the
+      pretty-printing engine that proposes to break the line here, if it is
+      necessary to properly print the rest of the material.
+      Otherwise, the pretty-printing engine never break lines (except
+      "in case of emergency" to avoid very bad output).
```

Comment:
Opening a box after the maximum indentation limit also induces breaks the line. It is a far too common situation to speak about emergency.